src/platform/qt/ShaderSelector.cpp

```diff
@@ -243,58 +245,85 @@ void ShaderSelector::addUniform(QGridLayout* settings, const QString& section, c
-			addUniform(settings, section, name + "[2]", &uniform->value.fvec3[2], uniform->min.fvec3[2], uniform->max.fvec3[2], 0, 2);
-			break;
-		case GL_FLOAT_VEC4:
-			addUniform(settings, section, name + "[0]", &uniform->value.fvec4[0], uniform->min.fvec4[0], uniform->max.fvec4[0], 0, 0);
-			addUniform(settings, section, name + "[1]", &uniform->value.fvec4[1], uniform->min.fvec4[1], uniform->max.fvec4[1], 0, 1);
-			addUniform(settings, section, name + "[2]", &uniform->value.fvec4[2], uniform->min.fvec4[2], uniform->max.fvec4[2], 0, 2);
-			addUniform(settings, section, name + "[3]", &uniform->value.fvec4[3], uniform->min.fvec4[3], uniform->max.fvec4[3], 0, 3);
-			break;
-		case GL_INT:
-			addUniform(settings, section, name, &uniform->value.i, uniform->min.i, uniform->max.i, 0, 0);
-			break;
-		case GL_INT_VEC2:
-			addUniform(settings, section, name + "[0]", &uniform->value.ivec2[0], uniform->min.ivec2[0], uniform->max.ivec2[0], 0, 0);
-			addUniform(settings, section, name + "[1]", &uniform->value.ivec2[1], uniform->min.ivec2[1], uniform->max.ivec2[1], 0, 1);
-			break;
-		case GL_INT_VEC3:
-			addUniform(settings, section, name + "[0]", &uniform->value.ivec3[0], uniform->min.ivec3[0], uniform->max.ivec3[0], 0, 0);
-			addUniform(settings, section, name + "[1]", &uniform->value.ivec3[1], uniform->min.ivec3[1], uniform->max.ivec3[1], 0, 1);
-			addUniform(settings, section, name + "[2]", &uniform->value.ivec3[2], uniform->min.ivec3[2], uniform->max.ivec3[2], 0, 2);
-			break;
-		case GL_INT_VEC4:
-			addUniform(settings, section, name + "[0]", &uniform->value.ivec4[0], uniform->min.ivec4[0], uniform->max.ivec4[0], 0, 0);
-			addUniform(settings, section, name + "[1]", &uniform->value.ivec4[1], uniform->min.ivec4[1], uniform->max.ivec4[1], 0, 1);
-			addUniform(settings, section, name + "[2]", &uniform->value.ivec4[2], uniform->min.ivec4[2], uniform->max.ivec4[2], 0, 2);
-			addUniform(settings, section, name + "[3]", &uniform->value.ivec4[3], uniform->min.ivec4[3], uniform->max.ivec4[3], 0, 3);
-			break;
+	if (defaultPage) {
+		addMatchingUniformRows(shader,layout,name,pass,"",true);
+	} else {
+		std::ifstream input(m_shaderPath.toStdString()+"/manifest.ini");
+		for( std::string line; getline( input, line ); ){
+			if (line.rfind("[pass.",0) == 0) {
+				size_t passDigits = line.substr(6,std::string::npos).find(".");
+				if (passDigits != std::string::npos) {
+					std::string passString = line.substr(6,passDigits);
+					int uniformPass = std::stoi(passString);
+					if (pass == uniformPass) {
+						size_t uniformEnd = line.find("]");
+						if (uniformEnd != std::string::npos) {
+							size_t uniformStart = 6+passDigits+9;
+							std::string uniformName = line.substr(uniformStart,uniformEnd-uniformStart);
+							addMatchingUniformRows(shader,layout,name,pass,uniformName,false);
+						}
+					}
+				}
+			}
 		}
```

Comment:
This is Qt code; use Qt classes like QFile and QString. Do not use STL classes. Please also follow style conventions.